Source/WebCore/css/ShorthandSerializer.cpp

```diff
@@ -1171,20 +1171,24 @@ String ShorthandSerializer::serializeWhiteSpace() const
 {
     auto whiteSpaceCollapse = longhandValueID(0);
     auto textWrap = longhandValueID(1);
-    if (whiteSpaceCollapse == CSSValueBreakSpaces && textWrap == CSSValueWrap)
-        return nameString(CSSValueBreakSpaces);
+
+    // Convert to backwards-compatible keywords if possible.
     if (whiteSpaceCollapse == CSSValueCollapse && textWrap == CSSValueWrap)
         return nameString(CSSValueNormal);
-    if (whiteSpaceCollapse == CSSValueCollapse && textWrap == CSSValueNowrap)
-        return nameString(CSSValueNowrap);
     if (whiteSpaceCollapse == CSSValuePreserve && textWrap == CSSValueNowrap)
         return nameString(CSSValuePre);
-    if (whiteSpaceCollapse == CSSValuePreserveBreaks && textWrap == CSSValueWrap)
-        return nameString(CSSValuePreLine);
     if (whiteSpaceCollapse == CSSValuePreserve && textWrap == CSSValueWrap)
         return nameString(CSSValuePreWrap);
+    if (whiteSpaceCollapse == CSSValuePreserveBreaks && textWrap == CSSValueWrap)
+        return nameString(CSSValuePreLine);
 
-    return String();
+    // Omit default longhand values.
+    if (whiteSpaceCollapse == CSSValueCollapse)
+        return nameString(textWrap);
+    if (textWrap == CSSValueWrap)
+        return nameString(whiteSpaceCollapse);
+
+    return makeString(nameString(whiteSpaceCollapse), ' ', nameString(textWrap));
```

Comment:
These should use `nameLiteral`.

Reply:
Oh sorry, @nt1m, I was way too terse. I meant that two cases here that are arguments to `makeString` should use `nameLiteral` for better efficiency, since appending an `ASCIILiteral` can be more efficient than doing the same with a `String`. All the ones that return a value to a `String` should use `nameString` for better efficiency, using the existing string rather than creating a new one.